### PR TITLE
[UPDATE] memoize.md

### PR DESCRIPTION
**Pull request by atomiks**
## Description

Uses a `Map` instead of a plain object to store any value as a key, and also allows access to the cache by setting a `.cache` property on the returned function.

## What does your PR belong to?
- [ ] Website
- [x] Snippets
- [ ] General / Things regarding the repository (like CI Integration)

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (non-breaking improvement of a snippet)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have checked that the changes are working properly
- [x] I have checked that there isn't any PR doing the same
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: snippets/memoize.md [hunk 1]
```diff
@@ -2,13 +2,16 @@
...
 Return a function which takes a single argument to be supplied to the memoized function by first checking if the function's output for that specific input value is already cached, or store and return it if not.
+Allow access to the `cache` by setting it as a property on the returned function.
 
 ```js
 const memoize = fn => {
-  const cache = Object.create(null);
-  return value => cache[value] || (cache[value] = fn(value));
+  const cache = new Map();
+  const cached = val => cache.has(val) ? cache.get(val) : (cache.set(val, fn(val)) && cache.get(val));
+  cached.cache = cache;
```
Comment (skatcat31) [inline snippets/memoize.md]: Why are we exposing the Cache? This leads to outside interference leading to unpredictable results.. We should keep it a private should we not?

**Comment by atomiks** [comment]
Lodash exposes the `cache` property.


>Note: The cache is exposed as the cache property on the memoized function. Its creation may be customized by replacing the _.memoize.Cache constructor with one whose instances implement the Map method interface of clear, delete, get, has, and set.

**Comment by atomiks** [comment]
It doesn't make sense for memoized functions to be variadic, right? 

Unless you pass it a single object or array (the SAME object/array -- otherwise you can check if the new one has the exact same elements/props as the stored one...)

**Comment by skatcat31** [comment]
Memoized functions need to be unary from what I know

**Pull request merged**